courseflow component under ontrack profile section

Knightwau2 opened the pull request:
_Any italic text should be deleted from the final Pull Request text, including this line_

before:
![before](https://github.com/thoth-tech/doubtfire-web/assets/101180020/db40fe0f-5e5c-4c78-a128-96b002504c23)
after:
![after1](https://github.com/thoth-tech/doubtfire-web/assets/101180020/2a324050-970e-4bad-8ba8-d854294f16f1)


# Description

added courseflow component at doubtfire-web/src/app/courseflow, router courseflow component at header, profile section, in the third button of drop down menu. 
there isnt any logos for courseflow yet, so it is just a button with text that links to courseflow component.

Fixes # (issue)

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

i test it by logging into ontrack server, enter in student/aconvenor account and navigate to the user profile section, drop down menu to see if the courseflow component exists there.
once successful, you should see a button named courseflow as the third option under profile drop-down menu.

## Testing Checklist:

- [x] Tested in latest Chrome
- [x] Tested in latest Safari
- [x] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have requested a review from @macite and @jakerenzella on the Pull Request

macite committed:
fix: ensure task submission status updated on init

macite committed:
fix: correct broken learning outcome alignment text

macite committed:
chore(release): 7.0.16

macite committed:
Merge branch '7.0.x' into development

macite committed:
Merge branch 'add-tii-integration' of https://github.com/doubtfire-lms/doubtfire-web into add-tii-integration

macite committed:
fix: ensure accepted TII eula details are retained

maddernd committed:
refactor: change-the-wording-of-fail-tasks

changed the wording on tasks from fail to needs improvement

refactor

macite committed:
fix: require eula for any user

macite committed:
Merge pull request #772 from maddernd/quality/change-fail-text

fix: change-the-wording-of-fail-tasks

macite committed:
chore(release): 7.0.17

Commit:
refactor(migration): sign-in component migration replaced fxlayout with tailwind equivalent

Commit:
refactor(migration): fixed wordmark responsiveness

Commit:
refactor(migration): fixed wordmark responsiveness

Commit:
refactor(migration): fix minor errors after peer review

jakerenzella committed:
refactor: (WIP) add new all units

jakerenzella committed:
fix: update package lock

jakerenzella committed:
build: update deps

jakerenzella committed:
Angular CLI update for packages - @angular/cdk@17, @angular/material@17

jakerenzella committed:
refactor: remove showdown

jakerenzella committed:
refactor: update marked

jakerenzella committed:
refactor: upgrade eslint

macite committed:
fix: ensure teaching staff loaded before units in a teaching period

macite committed:
Merge branch '7.0.x' into add-tii-integration

macite committed:
fix: ensure logout in timeout always signs out

macite committed:
Merge branch 'add-tii-integration' of https://github.com/doubtfire-lms/doubtfire-web into add-tii-integration

macite committed:
feat: add ability to open turnitin viewer

macite committed:
fix: reinstate break model in teaching period editor

macite committed:
chore(release): 7.0.18

jakerenzella committed:
build: upgrade uirouter

jakerenzella committed:
Angular CLI update for packages - @angular/core@17, @angular/cli@17

jakerenzella committed:
Angular CLI update for packages - @angular/cdk@17, @angular/material@17

TinyServal committed:
fix: add the correct enterkeyhint to the comment composer

macite committed:
fix: ensure similarities without files still view

macite committed:
Merge branch 'development' into add-tii-integration

macite committed:
chore: update entity service

macite committed:
feat: show TII accepted status in profile

macite committed:
Merge branch 'development' into add-tii-integration

macite committed:
chore(release): 8.0.0-0

macite committed:
fix: update deploy dockerfile

macite committed:
chore(release): 8.0.0-1

macite committed:
fix: correct npm install for dev container image

macite committed:
fix: ensure npm will force install

macite committed:
Merge branch 'add-tii-integration' into 8.0.x

macite committed:
fix: align center breaks

macite committed:
chore(release): 7.0.19

TinyServal committed:
refactor: remove unused task-status-selector module

TinyServal committed:
refactor: remove unused task-definition-selector component

TinyServal committed:
refactor: remove unused progress-modal module

TinyServal committed:
refactor: remove unused tutor-group-manager component

TinyServal committed:
refactor: remove unused task-outcomes-card component

TinyServal committed:
fix: remove the incorrect option from for scrollIntoView()

"top" is not a valid option, "start" is the default so there is no need
to define it explicitly

jakerenzella committed:
fix: remove flex-end from tcv

jakerenzella committed:
Merge branch '7.0.x' of https://github.com/doubtfire-lms/doubtfire-web into 7.0.x

macite committed:
fix: ensure sign out on timeout even with token delete error

macite committed:
chore(release): 7.0.20

jakerenzella committed:
fix: fix teaching periods service keys

macite committed:
Merge pull request #825 from jakerenzella/fix/update-teaching-period

fix: fix teaching periods service keys

macite committed:
chore(release): 7.0.21-0

TinyServal committed:
feat: accept promela files

macite committed:
chore(release): 7.0.21

jakerenzella committed:
build: upgrade uirouter hybrid

jakerenzella committed:
feat: remove misleading progress statement

jakerenzella committed:
feat: remove misleading progress statement

macite committed:
Merge pull request #828 from jakerenzella/7.0.x

feat: remove misleading progress statement

jakerenzella committed:
fix: remove ngClass

jakerenzella committed:
build: upgrade v17

jakerenzella committed:
build: switch to community flex-layout

jakerenzella committed:
Revert "fix: remove ngClass"

This reverts commit d1d8c72ace21551a68bd4d0a8ba88eabb2ebdb9d.

jakerenzella committed:
fix: typo

jakerenzella committed:
Merge pull request #826 from ublefo/new/promela

feat: accept promela files

jakerenzella committed:
Merge pull request #824 from ublefo/quality/scrollIntoView

fix: remove the incorrect option from for scrollIntoView()

jakerenzella committed:
Merge pull request #823 from ublefo/refactor/module-cleanup

refactor: module cleanup

jakerenzella committed:
Merge pull request #827 from ublefo/fix/comment-button

fix: add the correct enterkeyhint to the comment composer

jakerenzella committed:
Merge pull request #782 from BrianJCal99/looks/migrate-sign-in-component

Looks/migrate sign in component

macite committed:
Merge branch 'development' of https://github.com/doubtfire-lms/doubtfire-web into development

macite committed:
Merge branch 'development' into 8.0.x

macite committed:
fix: correct change detection on task def

TinyServal committed:
fix: indicate mandatory fields in upload-submission-modal

jakerenzella committed:
Merge branch 'development' into new/add-new-all-units

jakerenzella committed:
feat: migrate unit list in all locations

jakerenzella committed:
refactor: remove old admin/unit states

TinyServal committed:
refactor: improve error messages, switch to date picker in extension-modal

jakerenzella committed:
fix: fix firefox-specific bug

jakerenzella committed:
fix: remove unused loading from f-units

TinyServal committed:
fix: add max length validator and change reason field to text-area

TinyServal committed:
fix: use tailwindcss classes

jakerenzella committed:
build: upgrade packages

macite committed:
fix: remove plagiarism checks

macite committed:
chore(release): 8.0.0-2

jakerenzella committed:
fix: reenable eula html fetch

jakerenzella committed:
Merge pull request #836 from jakerenzella/new/add-new-all-units

Migrate all unit lists

macite committed:
chore(release): 8.0.0-3

TinyServal committed:
fix: grade and quality point display on task-assessment-card

TinyServal committed:
chore: ignore nx cache and intellij config

TinyServal committed:
refactor: move utility functions from component to task model

Knightwau2 committed:
courseflow component under ontrack profile section

macite committed:
fix: ensure eula state remains visible

TinyServal committed:
build: add new build target for reverse proxy in devcontainer

TinyServal committed:
fix: correct filenames for grotesk fonts

jakerenzella committed:
feat: add task badge

jakerenzella committed:
Merge branch 'development' of https://github.com/doubtfire-lms/doubtfire-web into development

jakerenzella committed:
fix: ensure unit admin shows all units

jakerenzella committed:
fix: fix course progress bar on firefox

jakerenzella committed:
refactor: fix home page bugs

jakerenzella committed:
fix: fix margin issue on dropdown

jakerenzella committed:
Merge pull request #842 from jakerenzella/refactor/cleanup-home

refactor: fix home page bugs

jakerenzella committed:
fix: fix incorrect projcets listed in dropdown

jakerenzella committed:
Merge branch 'development' of https://github.com/doubtfire-lms/doubtfire-web into refactor/cleanup-home

jakerenzella committed:
Merge pull request #841 from ublefo/grotesk-font-fix

TinyServal committed:
fix: remove unused imports in extension-modal.service

TinyServal committed:
fix: only check grade if it exists

TinyServal committed:
fix: address linter issues in task.ts

jakerenzella committed:
Merge pull request #837 from ublefo/extension-request-refactor

Extension modal refactor

jakerenzella committed:
Merge pull request #839 from ublefo/task-assessment-card-fix

fix: grade and quality point display on task-assessment-card

jakerenzella committed:
fix: (WIP) fix alert service migration

jakerenzella committed:
refactor: migrate to new alert service

jakerenzella committed:
Merge branch 'development' into fix/refactor-alert-service

jakerenzella committed:
Merge pull request #843 from jakerenzella/fix/refactor-alert-service

Fix/refactor alert service

macite committed:
fix: copy dist browser to nginx html

macite committed:
chore(release): 8.0.0-4

Knightwau2 committed:
remove .nx and added CF logo

small update to CourceFlow component

macite committed:
chore: reset font

macite committed:
Merge branch 'development' of https://github.com/doubtfire-lms/doubtfire-web into development

macite committed:
Merge branch '7.0.x' into development

macite committed:
Merge branch 'development' into 8.0.x

macite committed:
chore(release): 8.0.0-5

macite committed:
chore: update github action versions

macite committed:
chore: ensure @nx package optionally installed

macite committed:
chore(release): 8.0.0-6

TinyServal committed:
fix: add missing imports

jakerenzella committed:
Merge pull request #844 from ublefo/fix-missing-imports

DudeUnleashed commented:
Tested and all works on my end, courseflow button now leads to new page on the site as expected.

jakerenzella committed:
Merge pull request #840 from ublefo/devcontainer-reverse-proxy

Knightwau2 committed:
small fix on UIrouter

make sure that ontrack button redirects to courseflow, and fix small issue when resolving conflicts

macite committed:
fix: sort units desc by start date

jakerenzella committed:
feat: add multi-badge support

macite committed:
fix: change time exceeded to feedback exceeded

macite committed:
fix: correct task resource upload actions

macite committed:
fix: zip upload for overseer

macite committed:
fix: hide overseer settings if automation disabled

macite committed:
fix: ensure unit load updates unit object if it already exists

DudeUnleashed commented:
All works again after small fix

jakerenzella committed:
feat: add working unit-badge

jakerenzella committed:
feat: add working unit-badge

jakerenzella committed:
Merge branch 'development' of https://github.com/doubtfire-lms/doubtfire-web into new/multi-badge-support

jakerenzella committed:
refactor: simplify unit code component

jakerenzella committed:
feat: add synchronised multi-badge

macite committed:
Merge branch '8.0.x' of https://github.com/doubtfire-lms/doubtfire-web into 8.0.x

jakerenzella committed:
Merge branch 'development' into 8.0.x

jakerenzella committed:
feat: qol improvements for unit codes

macite committed:
fix: slide unit code off left side

macite committed:
refactor: remove unused tailwind animation config

jakerenzella committed:
feat: small fixes to home and unit code

macite committed:
chore: allow _ before unused parameters

macite committed:
fix: ensure loading state changed after projects and unit roles load

macite committed:
fix: ensure staff task list checks unit exists

macite committed:
chore(release): 8.0.0

jakerenzella committed:
fix: fix splashscreen

jakerenzella committed:
fix: fix critical bugs

jakerenzella committed:
Merge branch '8.0.x' of https://github.com/doubtfire-lms/doubtfire-web into 8.0.x

keraan committed:
feat: implement live PDF search that updates dynamically as you type

jakerenzella committed:
Merge pull request #849 from keraan/enhance/allow-live-pdf-search

enhance/allow-live-pdf-search

macite committed:
chore(release): 8.0.1

jakerenzella committed:
fix: gradeservice bugs

jakerenzella committed:
fix: gradeservice bugs

jakerenzella committed:
Merge pull request #850 from jakerenzella/fix/fix-grade-service

Fix/fix grade service

jakerenzella committed:
chore(release): 8.0.2

jakerenzella committed:
feat: add 4-character abbreviations

jakerenzella committed:
chore(release): 8.0.3

jakerenzella committed:
fix: fix various issues with admin unit list

jakerenzella committed:
chore(release): 8.0.4

jakerenzella committed:
fix: ensure (click) is on button, not mat-icon

jakerenzella committed:
chore(release): 8.0.5

jakerenzella committed:
fix: ensure all admin units are fully loaded

jakerenzella committed:
chore(release): 8.0.6

jakerenzella committed:
fix: add task description editing

jakerenzella committed:
chore(release): 8.0.7

Emilecross committed:
fix: change grade values, change grade display

Changed grade values to index from 0..4 from -1..3

Emilecross committed:
fix: revert index changes, html changes for grade

revert grade indexing, edit html files to work with original indexing

Bahnschrift committed:
fix: inbox search box width

jakerenzella committed:
Merge pull request #852 from Bahnschrift/fix-inbox-search-width

keraan committed:
feat: added percentage to tooltip when hovering student's task progress
bar in the footer

Bahnschrift committed:
fix: task list scrolling

Bahnschrift committed:
fix: use tailwind classes instead of css styling

jakerenzella committed:
Merge pull request #855 from Bahnschrift/fix-inbox-scrolling

fix: task list scrolling

jakerenzella committed:
Merge pull request #854 from keraan/enhance/view-completion-percentage-on-hover

 enhance/view-completion-percentage-on-hover

jakerenzella committed:
Merge pull request #851 from Emilecross/fix/portfolio-grade-service-issues

fix/portfolio grade service issues

Emilecross committed:
fix: remove pdf viewer on task view when no sheet

change so when no tasksheet on tutor task view dont request for pdf

Emilecross committed:
fix: change from hiding element to not loading

change to load conditionally instead of hidden conditionally

Emilecross committed:
refactor: simplify condition for pdf-viewer load

change from two conditions to one using optional chaining

jakerenzella committed:
Merge pull request #856 from Emilecross/fix/hide-pdf-no-sheet

Fix/hide pdf no sheet

macite committed:
fix: ensure you can create new users via ui

macite committed:
feat: add auditor option to user

jakerenzella committed:
fix: fix task inbox shortcuts

jakerenzella committed:
Merge branch '8.0.x' of https://github.com/doubtfire-lms/doubtfire-web into 8.0.x

jakerenzella committed:
chore(release): 8.0.8

macite committed:
feat: ensure auditor can view unit list

macite committed:
refactor: remove if role attribute

macite committed:
fix: use transitions to authorise page access

macite committed:
feat: authorise auditor to access staff pages

macite committed:
feat: auditor home is the unit list

macite committed:
feat: enable admin to see all unit pages

macite committed:
Merge branch '7.0.x' of https://github.com/doubtfire-lms/doubtfire-web into 7.0.x

macite committed:
chore(release): 7.0.22

keraan committed:
fix: comment out fix and complete hotkeys temporarily

jakerenzella committed:
Merge pull request #861 from keraan/fix/marking-hotkeys

fix/marking-hotkeys

jakerenzella committed:
chore(release): 8.0.9

macite committed:
Merge branch '8.0.x' into development

macite committed:
fix: update entity service to correct portfolio progress load

macite committed:
fix: tidy up portfolio marking page to avoid save issues

macite committed:
chore(release): 7.0.23

macite committed:
Merge branch '7.0.x' into development

macite committed:
fix: grade icon for portfolio page

macite committed:
Merge branch '7.0.x' into development

macite committed:
chore: update package lock

macite committed:
fix: show burndown and pie chart on portfolio page

aNebula commented:
@Knightwau2 Please open up an upstream PR against the 'development' branch on https://github.com/doubtfire-lms/doubtfire-web , so that it can be merged for your personal github portfolio.
